include/nodes/internal/Converter.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "NodeData.hpp"
+
+#include <functional>
+#include <tuple>
+
+namespace QtNodes
+{
+
+using SharedNodeData = std::shared_ptr<NodeData>;
+
+// a function taking in NodeData and returning NodeData
+using Converter =
+  std::function<SharedNodeData(SharedNodeData)>;
```

Comment:
I think that `TypeConverter` might be a better name. My first thought when I see `Converter` is: "What is this converting?" Maybe `DataConverter`?